topic4.user/ProxyService.java

```diff
@@ -0,0 +1,40 @@
+package bootcamp412.topic4_user;
+
+import java.awt.image.BufferedImage;
+
+public class ProxyService implements Service {
+	private Service service;
+
+	public ProxyService() {
```

Comment:
[High] You have to add a Service as a parameter in the constructor method. In that case you are forcing to that service to be always an UserService.
http://www.makinggoodsoftware.com/2009/11/17/how-to-create-services-in-java/